### PR TITLE
Api documentation update

### PR DESCRIPTION
**Pull request by psspavan96**
API Documentation Update

/assign @ckadner

### CODE REVIEW

File: api/README.md [hunk 1]
```diff
@@ -4,20 +4,26 @@ An extension to the Kubeflow Pipeline API for Components and Models
...
 
-## Deploy to Kubernetes
+![This is an image](https://github.com/psspavan96/mlx/blob/API_Documentation_update/api/API_Codegen_Workflow.png)
 
-    kubectl apply -f ./server/mlx-api.yml
+The generation of Python code starts from Swagger Spec. The generated code has 2 outputs: the `/api/client` and the `/api/server` packages. There is no static HTML documentation that gets generated. There are also no Python resources with Swagger annotations from which we generate our Swagger spec, instead the Swagger UI will be generated on the fly when the Python server is started. There are however some docs under the client package, but those are in Markdown format which works well when browsing through the Github repo. The `/api/client/docs` contains some examples from which we create the Python scripts in the examples package. Once the server code is generated we need to copy any new API method stubs (if any new) from the `/api/server/swagger_server/controllers`  folder to the `/api/server/swagger_server/controllers_impl` folder and actually  write the business logic. If existing API method signatures got updated, we need to update the existing controller methods respectively.
```
Comment (ckadner) [inline api/README.md:11]: I think it would be easier to read if this paragraph is broken into bullet points

Comment (psspavan96) [reply api/README.md:11]: - The generation of Python code starts from Swagger Spec. The generated code has 2 outputs: the `/api/client` and the `/api/server` packages. 
- No static HTML documentation gets generated from Swagger Spec. There are no Python resources with Swagger annotations from which we generate our Swagger spec, instead the Swagger UI will be generated on the fly when the Python server is started. There are some docs under the client package, but those are in Markdown format which works well when browsing through the Github repo. 
- The `/api/client/docs` contains some examples from which we create the Python scripts in the examples package. 
- Once the server code is generated we need to copy any new API method stubs (if any new) from the `/api/server/swagger_server/controllers`  folder to the `/api/server/swagger_server/controllers_impl` folder and actually  write the business logic. If existing API method signatures got updated, we need to update the existing controller methods respectively.

Comment (psspavan96) [reply api/README.md:11]: Is the flow better?

Comment (ckadner) [reply api/README.md:11]: Yup, I just did a small edit in your comment above. Please update the `/api/README.md` file accordingly

**Pull request closed without merging**